helpers/theme-utils.ts

```diff
@@ -0,0 +1,23 @@
+import { Accent } from 'degen/dist/types/tokens';
+
+export type ThemeAccent = Accent | 'foreground';
+
+export const accentSequence: ThemeAccent[] = [
+  'orange',
+  'yellow',
+  'blue',
+  'green',
+  'indigo',
+  'pink',
+  'purple',
+  'red',
+  'teal',
+  'foreground'
+];
+
+//returns sequence in form of map {'orange': 'yellow', 'yellow': 'blue', ...}
+export const nextAccentMap = accentSequence.reduce((acc, color, index) => {
+  const isLastColor = index + 1 === accentSequence.length;
+  acc[color] = isLastColor ? accentSequence[0] : accentSequence[index + 1];
+  return acc;
+}, {} as Record<ThemeAccent, ThemeAccent>);
```

Comment:
Can also hardcode this as a switch/case as is done here: https://github.com/mirror-xyz/degen/blob/main/docs/src/components/ThemeSwitcher.tsx#L53 but I did it this way so that in the future it's easier to change the color sequence if that ever comes up.